lib/wallet/walletdb.js

```diff
@@ -207,67 +219,27 @@ class WalletDB extends EventEmitter {
 
     this.primary = wallet;
 
-    await this.migrateChange();
+    if (migrationResult.rescan) {
+      this.logger.info('Rescanning...');
+      await this.scan();
```

Comment:
I'm not sure but we might want to rescan from genesis here (`this.scan(0)`) In walletDB state there's this `startHeight` property but I think it's set every TX or something?

Reply:
Hm.. this follows old `migration` code.

`startHeight` is initial height of the walletDB initialization, it does not ever increase. It can decrease though. (e.g. rollback)

Reply:
As discussed I think this should be `scan(0)` and I'll open an issue about `startHeight` in general.